tensorflow/stream_executor/cuda/cuda_driver.cc

```diff
@@ -584,7 +587,8 @@ bool DeviceOptionsToContextFlags(DeviceOptions device_options, int *flags) {
     // TODO(leary) Need to see if NVIDIA can expunge the leakiness in their
     // context creation: see http://b/13248943
 
-    res = dynload::cuCtxCreate_v2(&new_context, flags, device);
+    res = dynload::cuDevicePrimaryCtxSetFlags(device, flags);
```

Comment:
This looks great, but I think these primary context functions are only available in CUDA 7.5 and newer, and unfortunately one of the major TensorFlow users is currently stuck on CUDA 7.0. Could we check the libcuda version here and use the old calls for 7.0 and earlier?